File: src/main/java/com/timepath/hl2/hudeditor/CustomAction.java
package com.timepath.hl2.hudeditor;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.event.ActionEvent;

/**
 * @author TimePath
 */
public abstract class CustomAction extends AbstractAction {

    CustomAction(@NotNull String s, int mnemonic, KeyStroke shortcut, Icon icon) {
        super(Main.getString(s), icon);
        putValue(Action.MNEMONIC_KEY, mnemonic);
        putValue(Action.ACCELERATOR_KEY, shortcut);
    }

    CustomAction(@NotNull String s, int mnemonic, KeyStroke shortcut) {
        this(s, mnemonic, shortcut, null);
    }

    CustomAction(@NotNull String s, int mnemonic) {
        this(s, mnemonic, null);
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        action(e);
    }

    abstract void action(ActionEvent e);
}

File: src/main/kotlin/com/timepath/hl2/hudeditor/CustomAction.kt
package com.timepath.hl2.hudeditor


import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.Action
import javax.swing.Icon
import javax.swing.KeyStroke

public abstract class CustomAction(s: String, mnemonic: Int, shortcut: KeyStroke? = null, icon: Icon? = null)
: AbstractAction(Main.getString(s), icon) {

    init {
        putValue(Action.MNEMONIC_KEY, mnemonic)
        putValue(Action.ACCELERATOR_KEY, shortcut)
    }

    override fun actionPerformed(e: ActionEvent) {
        action(e)
    }

    abstract fun action(e: ActionEvent)
}
